src/main/kotlin/pl/put/smartgarden/infra/exception/SmartGardenException.kt

```diff
@@ -0,0 +1,34 @@
+package pl.put.smartgarden.infra.exception
+
+import org.springframework.http.HttpStatus
+
+/**
+ * Default Smart Garden exception, other new exceptions should extend this one.
+ */
+open class SmartGardenException : Exception {
```

Comment:
Może lepiej RuntimeException?

Reply:
Cóż początkowo chciałem żeby to był jednak "checked exception", ale widzę że Kotlin tego nie obsługuje 😄